lib/puppet/face/facts.rb

```diff
@@ -128,18 +128,31 @@
       summary _("Show legacy facts when querying all facts.")
     end
 
+    option("--value-only") do
+      summary _("Show only the value when the action is called with a single query")
+    end
+
     when_invoked do |*args|
       options = args.pop
 
       Puppet.settings.preferred_run_mode = :agent
       Puppet::Node::Facts.indirection.terminus_class = :facter
 
+      if options[:value_only] && !args.count.eql?(1)
+        raise ArgumentError, _("Incorrect use of --value-only argument, the argument must be used when querying for a single fact!")
+      end
```

Comment:
if `args` is `os`, then the fact group will produce multiple values, so I think we want to branch logic based on `facts.values.count`

Reply:
The code right now mimics facters behavior in the scenario when a single fact is requested:
```
~~ puppet facts os --value-only
{
  "architecture": "x86_64",
  "family": "Darwin",
  "hardware": "x86_64",
  "macosx": {
    "build": "18G95",
    "product": "Mac OS X",
    "version": {
      "full": "10.14.6",
      "major": "10.14",
      "minor": "6"
    }
  },
  "name": "Darwin",
  "release": {
    "full": "18.7.0",
    "major": "18",
    "minor": "7"
  }
}
```
```
~~ facter os
{
  architecture => "x86_64",
  family => "Darwin",
  hardware => "x86_64",
  macosx => {
    build => "18G95",
    product => "Mac OS X",
    version => {
      full => "10.14.6",
      major => "10.14",
      minor => "6"
    }
  },
  name => "Darwin",
  release => {
    full => "18.7.0",
    major => "18",
    minor => "7"
  }
} 
```
versus
```
~~ puppet facts os
{
  "os": {
    "architecture": "x86_64",
    "family": "Darwin",
    "hardware": "x86_64",
    "macosx": {
      "build": "18G95",
      "product": "Mac OS X",
      "version": {
        "full": "10.14.6",
        "major": "10.14",
        "minor": "6"
      }
    },
    "name": "Darwin",
    "release": {
      "full": "18.7.0",
      "major": "18",
      "minor": "7"
    }
  }
}
```
It feels like branching logic based on `facts.values.count` would only cause additional confusion and it's limiting the usage of this option.